static/js/settings.js

```diff
@@ -0,0 +1,63 @@
+(function() {
+	"use strict";
+
+    // TODO Find a way to make this reusable across the app?
+    function apiRequest(path, options) {
+        const cookieString = typeof document.cookie === "string" ? document.cookie : "";
+        const cookieStringArray = cookieString
+            .split(";")
+            .map(individualCookieString => individualCookieString.split("="))
+            .map(([cookieKey, cookieValue]) => [cookieKey.trim(), cookieValue.trim()]);
+        // Looks like the `argsIgnorePattern` option for ESLint doesn't like array destructuring:
+        // eslint-disable-next-line no-unused-vars
+        const [_csrfCookieKey, csrfCookieValue] = cookieStringArray.find(([cookieKey, _cookieValue]) => cookieKey === "csrftoken");
+        const headers = new Headers(options ? options.headers : undefined);
+        headers.set("X-CSRFToken", csrfCookieValue);
+        headers.set("Content-Type", "application/json");
+        return fetch(
+            `/api/v1${path}`,
+            {
+                mode: "same-origin",
+                ...options,
+                headers: headers,
+            },
+        );
+    }
+
+    const settingsForm = document.querySelector("#settings-form");
+    const labelCollectionCheckbox = document.querySelector("input#label-collection");
+    const labelCollectionOffWarning = document.querySelector("#label-collection-off-warning");
+    const saveError = document.querySelector("#save-error");
+
+    settingsForm.addEventListener("submit", async (event) => {
+        event.preventDefault();
+        const profileId = settingsForm.dataset.profileId;
+
+        const settings = {
+            server_storage: labelCollectionCheckbox.checked,
+        };
+
+        const response = await apiRequest(
+            `/profiles/${profileId}/`,
+            {
+                method: "PUT",
```

Comment:
[DRF wants clients to use `PATCH` HTTP request method to do "partial_update" actions](https://www.django-rest-framework.org/api-guide/routers/#api-guide) - i.e., where the client only sends the fields that it wants to update.

This doesn't matter yet, because the `PUT /profiles/{id}/` endpoint only accepts the single `server_storage` field right now anyway, but, if we keep this `PUT` request with this object like ...

```
{"server_storage": true}
```
... and then later we add the `subdomain` field to the API endpoint ... then this code will break because it's sending a `PUT` request without a value for `subdomain`.

Reply:
That's good to know! (/cc @maxxcrawford for visibility.) And I guess makes sense, in hindsight. 4ffa807424d26a6436718d1d184311ff3ad7daf8